### PR TITLE
Merge Raycasting into master

### PR DESCRIPTION
**Pull request by LeoMainini**
Fully formated

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,7 +58,7 @@ all: $(NAME)
 
 $(OBJS_DIR)/%.o : $(SRCS_DIR)/%.c | $(OBJS_DIR)
 	@$(ECHO) "$(GREEN)>>>>> Compiling $(RESET)$(notdir $<)$(GREEN) -> $(RESET)$(notdir $@)$(RESET)"
-	@gcc $(CFLAGS) -MMD -MP -c $(INCLUDES) $< -o $@
+	@gcc $(CFLAGS) $(O_FLAGS) -MMD -MP -c $(INCLUDES) $< -o $@
```
Comment (PukieDiederik) [inline Makefile:61]: Gcc needs to become $(CC)

File: include/cub3d.h [hunk 2]
```diff
@@ -15,6 +15,25 @@
...
+# include "vector.h"
+# include "options.h"
+
+enum {
+	ON_KEYDOWN = 2,
+	ON_DESTROY = 17,
+	L_ARROW = 65361,
+	R_ARROW = 65363,
+	ESC = 65307
+};
```
Comment (PukieDiederik) [inline include/cub3d.h:28]: Wasd keys need to be added.

**Review by LeoMainini — COMMENTED** [review]
All viewed by both

**Review by PukieDiederik — APPROVED** [review]
Looks good, just need a few changes in another pr.

**Pull request merged**